test/index.js

```diff
@@ -474,4 +475,156 @@ describe('HauteCouture', () => {
+
+        const noOptionsPlugin = (server, options, next) => {
+
+            HauteCouture()(server, (err) => {
+
+                server.app.realm = server.realm;
+                if (err) {
+                    return next(err);
+                }
+                next();
+            });
+        };
+
+        noOptionsPlugin.attributes = { name: 'no-options-plugin' };
+
+        server.register(noOptionsPlugin, (err) => {
+
+            if (err) {
+                return done(err);
+            }
+
+            // Just an example to show it used the caller's directory
+            expect(server.registrations['no-options-plugin']).to.exist();
+            done();
+        });
+    });
+
+    it('supports promises.', (done, onCleanup) => {
+
+        onCleanup((next) => {
+
+            reset();
+            return next();
+        });
+
+        const server = new Hapi.Server();
+        server.connection();
+
+        const plugin = (server, options, next) => {
+
+            HauteCouture()(server, options)
+            .then(() => {
+
+                server.app.realm = server.realm;
+                next();
+            })
+            .catch((err) => {
+
+                next(err);
+            });
+        };
+
+        plugin.attributes = { name: 'promise-plugin' };
+
+        server.register(plugin, (err) => {
+
+            if (err) {
+                return done(err);
+            }
+
+            // Just an example to show it used the caller's directory
+            expect(server.registrations['promise-plugin']).to.exist();
+            done();
+        });
+    });
+
+    it('allows options, next to be optional.', (done, onCleanup) => {
+
+        onCleanup((next) => {
+
+            reset();
+            return next();
+        });
+
+        const server = new Hapi.Server();
+        server.connection();
+
+        const noOptionsPlugin = (server, options, next) => {
+
+            HauteCouture()(server)
+            .then(() => {
+
+                server.app.realm = server.realm;
+                next();
+            })
+            .catch((err) => {
+
+                next(err);
+            });
+        };
+
+        noOptionsPlugin.attributes = { name: 'no-options-no-callback-plugin' };
+
+        server.register(noOptionsPlugin, (err) => {
+
+            if (err) {
+                return done(err);
+            }
+
+            // Just an example to show it used the caller's directory
+            expect(server.registrations['no-options-no-callback-plugin']).to.exist();
+            done();
+        });
+    });
+
+    it('promise returns error in catch.', (done, onCleanup) => {
+
+        onCleanup((next) => {
+
+            reset();
+            return next();
+        });
+
+        const server = new Hapi.Server();
+        server.connection();
+
+        const plugin = (server, options, next) => {
+
+            HauteCouture()(server)
+            .then(() => {
+
+                server.app.realm = server.realm;
```

Comment:
I was trying to pass an error to HauteCouture()(server, myError) because I'm missing code coverage on lines 24 and 25.

https://github.com/WilliamSWoodruff/haute-couture/blob/52135634b279cd4a55e09481eb4af5f4303e9483/lib/index.js#L24-L25